pennylane_lightning/src/tests/lightning_unittest.cpp

```diff
@@ -11,1066 +11,36 @@
-    auto output_state_vector_Y = vectorize(output_state_Y);
-    auto output_state_vector_Z = vectorize(output_state_Z);
-    auto output_state_vector_Rot = vectorize(output_state_Rot);
-
-    auto expected_vector_X = vectorize(expected_output_state_X);
-    auto expected_vector_Y = vectorize(expected_output_state_Y);
-    auto expected_vector_Z = vectorize(expected_output_state_Z);
-    auto expected_vector_Rot = vectorize(expected_output_state_Rot);
-
-    EXPECT_TRUE(expected_vector_X.isApprox(output_state_vector_X, tol));
-    EXPECT_TRUE(expected_vector_Y.isApprox(output_state_vector_Y, tol));
-    EXPECT_TRUE(expected_vector_Z.isApprox(output_state_vector_Z, tol));
-    EXPECT_TRUE(expected_vector_Rot.isApprox(output_state_vector_Rot, tol));
-}
-}  // namespace two_qubit_ops
-
-namespace three_qubit_ops {
-
-TEST(Toffoli, ApplyToAll) {
-    State_Xq<3> input_state_000(2, 2, 2);
-    input_state_000.setValues({{{1, 0}, {0, 0}}, {{0, 0}, {0, 0}}});
-    State_Xq<3> input_state_001(2, 2, 2);
-    input_state_001.setValues({{{0, 1}, {0, 0}}, {{0, 0}, {0, 0}}});
-    State_Xq<3> input_state_010(2, 2, 2);
-    input_state_010.setValues({{{0, 0}, {1, 0}}, {{0, 0}, {0, 0}}});
-    State_Xq<3> input_state_011(2, 2, 2);
-    input_state_011.setValues({{{0, 0}, {0, 1}}, {{0, 0}, {0, 0}}});
-    State_Xq<3> input_state_100(2, 2, 2);
-    input_state_100.setValues({{{0, 0}, {0, 0}}, {{1, 0}, {0, 0}}});
-    State_Xq<3> input_state_101(2, 2, 2);
-    input_state_101.setValues({{{0, 0}, {0, 0}}, {{0, 1}, {0, 0}}});
-    State_Xq<3> input_state_110(2, 2, 2);
-    input_state_110.setValues({{{0, 0}, {0, 0}}, {{0, 0}, {1, 0}}});
-    State_Xq<3> input_state_111(2, 2, 2);
-    input_state_111.setValues({{{0, 0}, {0, 0}}, {{0, 0}, {0, 1}}});
-
-    std::vector<State_Xq<3>> input_states{
-        input_state_000,
-        input_state_001,
-        input_state_010,
-        input_state_011,
-        input_state_100,
-        input_state_101,
-        input_state_110,
-        input_state_111,
-    };
-    std::vector<State_Xq<3>> output_states;
-
-    vector<int> w{0, 1, 2};
-
-    for (int i=0; i < 8; i++) {
-        output_states.push_back(contract_3q_op(input_states[i], "Toffoli", w));
-    }
-
-    State_Xq<3> target_state_110(2, 2, 2);
-    target_state_110.setValues({{{0, 0}, {0, 0}}, {{0, 0}, {0, 1}}});
-    State_Xq<3> target_state_111(2, 2, 2);
-    target_state_111.setValues({{{0, 0}, {0, 0}}, {{0, 0}, {1, 0}}});
-
-    auto expected_states = input_states;
-    expected_states[6] = target_state_110;
-    expected_states[7] = target_state_111;
-
-    for (int i=0; i < 8; i++) {
-        auto expected_vector = vectorize(expected_states[i]);
-        auto output_state = vectorize(output_states[i]);
-        EXPECT_TRUE(expected_vector.isApprox(output_state, tol));
+    for (size_t i = 0; i < inputs.size(); i++) {
+        size_t result = Pennylane::exp2(inputs[i]);
+        EXPECT_TRUE(result == outputs[i]);
```

Comment:
This is what the new file looks like:
https://github.com/PennyLaneAI/pennylane-lightning/blob/72c31a64b62878bde10c646ea33a5983bee95f10/pennylane_lightning/src/tests/lightning_unittest.cpp